exporter/fileexporter/file_exporter.go

```diff
@@ -59,55 +58,111 @@ func (e *fileExporter) consumeTraces(_ context.Context, td ptrace.Traces) error
 	if err != nil {
 		return err
 	}
-	buf = e.compressor(buf)
-	return e.exporter(e, buf)
+	_, err = e.exporter.Write(buf)
+	return err
 }
 
 func (e *fileExporter) consumeMetrics(_ context.Context, md pmetric.Metrics) error {
 	buf, err := e.metricsMarshaler.MarshalMetrics(md)
 	if err != nil {
 		return err
 	}
-	buf = e.compressor(buf)
-	return e.exporter(e, buf)
+	_, err = e.exporter.Write(buf)
+	return err
 }
 
 func (e *fileExporter) consumeLogs(_ context.Context, ld plog.Logs) error {
 	buf, err := e.logsMarshaler.MarshalLogs(ld)
 	if err != nil {
 		return err
 	}
-	buf = e.compressor(buf)
-	return e.exporter(e, buf)
+	_, err = e.exporter.Write(buf)
+	return err
+}
+
+type lineWriter struct {
+	mutex sync.Mutex
+	file  io.Writer
+}
+
+func NewLineWriter(cfg *Config, file io.Writer) io.Writer {
+	if cfg.Compression == "zstd" {
+		if fw, err := zstd.NewWriter(file); err == nil {
+			// flushing the compressed writer every second.
```

Comment:
Why is that and what happens if we get interrupted half a second in?

Reply:
> Why is that

Thanks for the review! The 1s flush interval is arbitrary. I decided to flush here otherwise flushing takes a while to occur. The flush interval allows flushing to occur sooner to reduce the overall size of file writes. But now moved everything to the original flush loop.

> what happens if we get interrupted half a second in

 I am now returning an io.WriteCloser which allows us to call Close() on the compressed writer. According to https://pkg.go.dev/github.com/klauspost/compress/zstd#Encoder.Close this will do a final flush, and once everything is written, the stream can be closed.